pandas/tseries/resample.py

```diff
@@ -673,18 +670,15 @@ def aggregate(self, arg, *args, **kwargs):
     def _get_new_index(self):
         """ return our new index """
         ax = self.ax
-        ax_attrs = ax._get_attributes_dict()
-        ax_attrs['freq'] = self.freq
-        obj = self._selected_obj
 
         if len(ax) == 0:
-            new_index = PeriodIndex(data=[], **ax_attrs)
-            return obj.reindex(new_index)
-
-        start = ax[0].asfreq(self.freq, how=self.convention)
-        end = ax[-1].asfreq(self.freq, how='end')
+            values = []
+        else:
+            start = ax[0].asfreq(self.freq, how=self.convention)
+            end = ax[-1].asfreq(self.freq, how='end')
```

Comment:
If we treat `PeriodIndex`s as an Interval-like Index, I think this should always be `how='start'` and `how='end'` for start & end respectively. Being able to specify these seems like misapplying the logic from `DatetimeIndex`.

So if you're upsampling, the whole super-period at the start would be included, and if you're downsampling, every sub-period would be included.

Making that change makes only [this](https://github.com/pydata/pandas/blob/master/pandas/tseries/tests/test_resample.py#L1834) test fail. I can't see the logic behind that test's expected - that just the final point of the first year would be wanted?

If people agree I'd be up for doing that, but probably counts as an API breakage

```python
Series length are different
[left]:  2191, PeriodIndex(['1990-01-01', '1990-01-02', '1990-01-03', '1990-01-04',
             '1990-01-05', '1990-01-06', '1990-01-07', '1990-01-08',
             '1990-01-09', '1990-01-10',
             ...
             '1995-12-22', '1995-12-23', '1995-12-24', '1995-12-25',
             '1995-12-26', '1995-12-27', '1995-12-28', '1995-12-29',
             '1995-12-30', '1995-12-31'],
            dtype='int64', length=2191, freq='D')
[right]: 1827, PeriodIndex(['1990-12-31', '1991-01-01', '1991-01-02', '1991-01-03',
             '1991-01-04', '1991-01-05', '1991-01-06', '1991-01-07',
             '1991-01-08', '1991-01-09',
             ...
             '1995-12-22', '1995-12-23', '1995-12-24', '1995-12-25',
             '1995-12-26', '1995-12-27', '1995-12-28', '1995-12-29',
             '1995-12-30', '1995-12-31'],
            dtype='int64', length=1827, freq='D')

```